File: vp-api/src/main/java/com/bastiansmn/vp/event/EventService.java
package com.bastiansmn.vp.event;

import com.bastiansmn.vp.event.dto.EventCreationDto;
import com.bastiansmn.vp.event.dto.EventUpdateDto;
import com.bastiansmn.vp.exception.FunctionalException;

import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface EventService {

    EventDAO fetchById(Long event_id) throws FunctionalException;

    EventDAO create(EventCreationDto event) throws FunctionalException;

    EventDAO updateDates(EventUpdateDto event);

    EventDAO participate(Long event_id) throws FunctionalException;

    EventDAO unparticipate(Long eventId) throws FunctionalException;

    void delete(Long event_id) throws FunctionalException;

    List<EventDAO> fetchAllOfProject(String projectId, String from, String to) throws FunctionalException;

    List<EventDAO> myEvents(String from, String to) throws FunctionalException;

    Boolean isParticipating(Long eventId) throws FunctionalException;

    Boolean isParticipating(EventDAO event);

    Boolean createdByMe(Long eventId) throws FunctionalException;

    Boolean createdByMe(EventDAO event);

}

File: vp-ui/src/app/services/event/event.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {Observable} from "rxjs";
import Event from "../../model/event.model";

@Injectable({
  providedIn: 'root'
})
export class EventService {

  constructor(
    private http: HttpClient
  ) { }

  getEventsFromProject(projectId: string, from: Date, to: Date): Observable<Event[]> {
    let params = new HttpParams();
    params = params.appendAll({
      projectId: projectId,
      from: from.toLocaleString(),
      to: to.toLocaleString()
    });

    return this.http.get<Event[]>("/api/v1/event/allOfProject", {params: params});
  }

  getMyEvents(from: Date, to: Date): Observable<Event[]> {
    let params = new HttpParams();
    params = params.appendAll({
      from: from.toISOString(),
      to: to.toISOString()
    });

    return this.http.get<Event[]>("/api/v1/event/myEvents", {params: params});
  }

  participate(event: Event): Observable<Event> {
    return this.http.put<Event>("/api/v1/event/participate?event_id=" + event.event_id, null);
  }

  unparticipate(event: Event): Observable<Event> {
    return this.http.put<Event>("/api/v1/event/unparticipate?event_id=" + event.event_id, null);
  }

  create(event: Event, project_id: string): Observable<Event> {
    return this.http.post<Event>("/api/v1/event/create", {
      project_id: project_id,
      name: event.name,
      description: event.description,
      videoCallLink: event.videoCallLink,
      date_start: event.date_start,
      date_end: event.date_end,
      whole_day: event.whole_day
    });
  }

  delete(event: Event): Observable<void> {
    return this.http.delete<void>("/api/v1/event/delete?event_id=" + event.event_id);
  }
}
